src/lib/sql/sql.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "types.hpp"
+
+#include "sql_pipeline.hpp"
+#include "sql_pipeline_statement.hpp"
+
+namespace opossum {
+
+class Optimizer;
+
+/**
+ * Interface for the configured execution of SQL.
+ *
+ * Minimal usage:
+ *      SQL{"SELECT * FROM t;"}.pipeline().get_result_table()
```

Comment:
To be honest, I am not a fan of that chained syntax. :/
It is mostly uncommon in C++ and it is unclear on what you are operating `SQL{...}.set_optimizer(...).pipeline().get_result_table()` has three objects that you operate on. We don‘t save anything by chaining it, except for maybe two line breaks.

Reply:
I have seen these chained setters to be prefixed with `with_`, so it would be `SQL{"..."}.with_mvcc_disabled().with_transaction_context(context).create_pipeline()`

Reply:
We save typing `sql_pipeline_builder.` over and over again and its respective linebreaks. I think not chaining this would feel reeeeaaally awkward. (and Java-esque)

Reply:
„code is read far more often than it is written“
https://softwareengineering.stackexchange.com/questions/80244/are-there-any-actual-drawbacks-to-self-referential-method-chaining

I find this harder to follow than one-thing-per-line. It gets worse when things span over multiple lines and the linter dows weird things.

Reply:
I'm also for option 1 :) I think this is a very common approach when using a builder.